src/applications/personalization/profile-2/components/Profile2Router.jsx

```diff
@@ -204,6 +232,8 @@ const mapStateToProps = state => {
   const isEligibleToSignUp = directDepositAddressIsSetUp(state);
   const is2faEnabled = isMultifactorEnabled(state);
   const shouldFetchDirectDepositInformation = isEvssAvailable && is2faEnabled;
+  const shouldShowMilitaryInformation =
+    selectProfile(state)?.veteranStatus?.servedInMilitary || false;
```

Comment:
Since we're not computing a value, I am not sure if having a selector would really be beneficial here other than the code looking a bit cleaner

Reply:
I'm a fan of selectors just because it abstracts away any required knowledge of the redux state. Even if it's super simple like this one here, it's nice to have this in a selector because:
- this line would be slightly more readable (compare this line with the optional chaining to the line above it: `const is2faEnabled = isMultifactorEnabled(state)`
- probably more importantly, if the structure of the state ever changed you've only have to change the selector rather than having to change all the instances where this same prop-drilling is done all over the codebase.

Anyway, no need to add that now!